packages/flutter/lib/src/widgets/inherited_model.dart

```diff
@@ -6,53 +6,62 @@ import 'dart:collection';
+/// model.
 ///
 /// An inherited widget's dependents are unconditionally rebuilt when the
-/// inherited widget changes per [InheritedWidget.updateShouldNotify].
-/// This widget is similar except that dependents aren't rebuilt
-/// unconditionally.
+/// inherited widget changes per [InheritedWidget.updateShouldNotify]. This
+/// widget is similar except that dependents aren't rebuilt unconditionally.
 ///
-/// Widgets that depend on an [InheritedModel] qualify their dependence
-/// with a value that indicates what "aspect" of the model they depend
-/// on. When the model is rebuilt, dependents will also be rebuilt, but
-/// only if there was a change in the model that corresponds to the aspect
-/// they provided.
+/// Widgets that depend on an [InheritedModel] qualify their dependence with a
+/// value that indicates what "aspect" of the model they depend on. When the
+/// model is rebuilt, dependents will also be rebuilt, but only if there was a
+/// change in the model that corresponds to the aspect they provided.
 ///
 /// The type parameter `T` is the type of the model aspect objects.
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=ml5uefGgkaA}
 ///
 /// Widgets create a dependency on an [InheritedModel] with a static method:
-/// [InheritedModel.inheritFrom]. This method's `context` parameter
-/// defines the subtree that will be rebuilt when the model changes.
-/// Typically the `inheritFrom` method is called from a model-specific
-/// static `of` method. For example:
+/// [InheritedModel.inheritFrom]. This method's `context` parameter defines the
+/// subtree that will be rebuilt when the model changes. Typically the
+/// `inheritFrom` method is called from a model-specific static `maybeOf` or
+/// `of` methods, a convention that is present in many Flutter framework classes
+/// which look things up. For example:
 ///
 /// ```dart
 /// class MyModel extends InheritedModel<String> {
 ///   const MyModel({super.key, required super.child});
+///
 ///   // ...
-///   static MyModel? of(BuildContext context, String aspect) {
+///   static MyModel? maybeOf(BuildContext context, String aspect) {
 ///     return InheritedModel.inheritFrom<MyModel>(context, aspect: aspect);
 ///   }
+///
+///   // ...
+///   static MyModel of(BuildContext context, String aspect) {
+///     final MyModel? result = maybeOf(context, aspect);
+///     assert(result != null, 'Unable to find an instance of MyModel...');
+///     return result!
+/// ;
+///   }
 /// }
 /// ```
 ///
-/// Calling `MyModel.of(context, 'foo')` means that `context` should only
-/// be rebuilt when the `foo` aspect of `MyModel` changes. If the aspect
-/// is null, then the model supports all aspects.
+/// Calling `MyModel.maybeOf(context, 'foo')` or `MyModel.maybeOf(context,
+/// 'foo')` means that `context` should only be rebuilt when the `foo` aspect of
+/// `MyModel` changes. If the aspect is null, then the model supports all
```

Comment:
Not really related to your change, but maybe you could clean this up while you're in here: The sentence about "null" here doesn't seem to make much sense given that "aspect" is typed as non-nullable in the example above? Should we just remove it? (or update the example to allow null, e.g. by making the aspect argument optional).

Reply:
Okay, I did that.  In the process, I wondered if the type of the model should be `InheritedModel<String?>` or `InheritedModel<String>`. I left it as the latter, since it seems like the API for `inheritFrom` takes an `Object?` no matter what the type is, and that's the more specific type. I think it doesn't actually matter though, right?

Reply:
I believe `InheritedModel<String>` is correct here.